lambdas/process-cri-callback/src/main/java/uk/gov/di/ipv/core/processcricallback/ProcessCriCallbackHandler.java

```diff
@@ -145,12 +145,17 @@ public APIGatewayProxyResponseEvent handleRequest(
         } catch (InvalidCriCallbackRequestException e) {
             if (e.getErrorResponse() == ErrorResponse.NO_IPV_FOR_CRI_OAUTH_SESSION) {
                 LOGGER.error(e.getErrorResponse(), e);
-                return ApiGatewayResponseGenerator.proxyJsonResponse(
+                var pageOutput = StepFunctionHelpers.generatePageOutputMap(
+                        "error",
                         HttpStatus.SC_UNAUTHORIZED,
-                        StepFunctionHelpers.generatePageOutputMap(
-                                "error",
-                                HttpStatus.SC_UNAUTHORIZED,
-                                PYI_TIMEOUT_RECOVERABLE_PAGE_ID));
+                        PYI_TIMEOUT_RECOVERABLE_PAGE_ID);
+                if (callbackRequest != null) {
+                    var criOAuthSessionItem = criOAuthSessionService.getCriOauthSessionItem(callbackRequest.getState());
+                    pageOutput.put("clientOAuthSessionId", criOAuthSessionItem.getClientOAuthSessionId());
+                    return ApiGatewayResponseGenerator.proxyJsonResponse(
```

Comment:
Can we move this return out of the if, so it always returns an error, just only includes the clientOAuthSessionId if available?